trieste/acquisition/rule.py

```diff
@@ -294,3 +294,61 @@ def acquire(
         state_ = TrustRegion.State(acquisition_space, eps, y_min, is_global)
 
         return point, state_
+
+
+class BatchAcquisitionRule(AcquisitionRule[None, SearchSpace]):
+    """ Implements a acquisition rule for a batch of query points """
+
+    def __init__(self, num_query_points: int,
+                 builder: Optional[AcquisitionFunctionBuilder] = None):
+        """
+        :param num_query_points: The number of points to acquire.
+        :param builder: The acquisition function builder to use.
+            :class:`EfficientGlobalOptimization` will attempt to **maximise** the corresponding
+            acquisition function. Defaults to :class:`~trieste.acquisition.ExpectedImprovement`
+            with tag `OBJECTIVE`.
+        """
+
+        if not num_query_points > 0:
+            raise ValueError(
+                f"Number of query points must be greater than 0, got {num_query_points}"
+            )
+
+        self._num_query_points = num_query_points
+
+        if builder is None:
+            builder = ExpectedImprovement().using(OBJECTIVE)  # this is a placeholder for now
+            # num_samples = 100
+            # builder = MonteCarloExpectedImprovement(eps_shape=[num_samples, num_query_points, 1]).using(OBJECTIVE)  # [S, B, L]
+
+        self._builder = builder
+
+    def _vectorize_batch_acquisition(self, acquisition_function):
+        return lambda at: acquisition_function(tf.reshape(at, at.shape[:-1].as_list() + [self._num_query_points, -1]))
+
+    def acquire(
+        self,
+        search_space: SearchSpace,
+        datasets: Mapping[str, Dataset],
+        models: Mapping[str, ProbabilisticModel],
+        state: None = None,
+    ) -> Tuple[QueryPoints, None]:
+        """
+        Return the batch of query points that optimizes the acquisition function produced by `builder` (see
+        :meth:`__init__`).
+        :param search_space: The global search space over which the optimization problem
+            is defined.
+        :param datasets: The known observer query points and observations.
+        :param models: The models of the specified ``datasets``.
+        :param state: Unused.
+        :return: The batch of points to query, and `None`.
+        """
+        expanded_search_space = search_space ** self._num_query_points
```

Comment:
this generates the following error during tests: 
Unsupported operand types for ** ("SearchSpace" and "int")

Reply:
that's cos `__pow__` is only defined on `Box`, not on all `SearchSpace`s. You can either restrict this rule to only work with `Box`es for now (with `class BatchAcquisitionRule(AcquisitionRule[None, Box]):` and
```
 def acquire(
        self,
        search_space: Box,
        ...
```
or you can implement `__pow__` on `DiscreteSearchSpace` too

Reply:
i could implement `__pow__` on `DiscreteSearchSpace` for you?

Reply:
I'd got for the latter - if you have a clear idea how to implement pow for discrete spaces, go for it! I don't know how to do it efficiently.